### PR TITLE
W5 - D1 - thanhhoang - RESTful with Retrofit and Gson

### PR DESCRIPTION
**Pull request by at-thanhhoang**
Task #51

### CODE REVIEW

**Comment by at-thanhhoang** [comment]
@at-trungnguyen . Nhờ anh review giúp em với. thanks anh.

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+    private void downloadImages() {
+        ApiUtils.getServiceDownload().downloadImages(ACCESS_TOKEN, 50).enqueue(new Callback<List<ImageInfo>>() {
+            @Override
+            public void onResponse(@NonNull Call<List<ImageInfo>> successCall, @NonNull Response<List<ImageInfo>> response) {
+                mUrlList.clear();
+                for (ImageInfo images : Objects.requireNonNull(response.body())) {
+                    mUrlList.add(images.getUrl());
+                }
+                mProgressDialog.dismiss();
```
[the remark this answers is not in the document]
Comment (at-thanhhoang) [reply app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java]: move đi đâu anh

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                            openGallery();
+                        }
+                    }
+                });
+        Dialog dialog = builder.create();
+        dialog.show();
+    }
+
+    private void openCamera() {
+        Intent cameraIntent = new Intent(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
```
Comment (at-dev-support-01) [inline app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'android.provider.MediaStore.ACTION_IMAGE_CAPTURE' due to existing import 'android.provider.MediaStore'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+                final Uri imageUri = data.getData();
+                uploadImage(imageUri);
+            }
+        }
+    }
+
+    private String getRealPathImage(Uri imageUri) {
+        String[] projection = {MediaStore.Images.Media.DATA};
+        Cursor cursor = getContentResolver().query(imageUri, projection, null, null, null);
+        if (cursor == null) return null;
```
Comment (at-dev-support-01) [inline app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java]: WARNING: [IfStmtsMustUseBraces] Avoid using if statements without curly braces
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/braces.html#IfStmtsMustUseBraces

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java (hunk 2 above)
Comment (at-dev-support-01) [inline app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'android.provider.MediaStore.ACTION_IMAGE_CAPTURE' due to existing import 'android.provider.MediaStore'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java (hunk 3 above)
Comment (at-dev-support-01) [inline app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/RestAPIActivity.java]: WARNING: [IfStmtsMustUseBraces] Avoid using if statements without curly braces
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/braces.html#IfStmtsMustUseBraces

**Comment by at-thanhhoang** [comment]
@at-trungnguyen em đã fix xong bài tập anh giao. Nhờ anh review lại giúp em với. thanks anh ^_^

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/adapter/RestAPIAdapter.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        private ImageView mImgResult;
+
+        private MyViewHolder(View itemView) {
+            super(itemView);
+            mImgResult = itemView.findViewById(R.id.imgResultAPI);
+            mImgResult.setOnClickListener(v -> mListener.onImageClick(getAdapterPosition()));
+        }
+    }
+
+    public interface OnClickImagesListener {
```
Comment (at-trungnguyen) [inline app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/adapter/RestAPIAdapter.java]: add `java doc`

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/models/ImageInfo.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
+package asiantech.internship.summer.rest_api.models;
+
+import com.google.gson.annotations.SerializedName;
+
+public class ImageInfo {
+    @SerializedName("image_id")
+    private String imageId;
+    @SerializedName("permalink_url")
+    private String permalinkUrl;
+    @SerializedName("url")
```
Comment (at-trungnguyen) [inline app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/models/ImageInfo.java]: delete this line, no need `@SerializedName("url")`

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/models/ImageInfo.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    private String url;
+    @SerializedName("type")
```
Comment (at-trungnguyen) [inline app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/models/ImageInfo.java]: delete `@SerializedName("type")`

File: app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/models/ImageInfo.java [hunk 7]
```diff
@@ -1,19 +1,11 @@
 package asiantech.internship.summer.rest_api.models;
 
-import com.google.gson.annotations.SerializedName;
-
 public class ImageInfo {
-    @SerializedName("image_id")
```
Comment (at-trungnguyen) [inline app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/models/ImageInfo.java]: this is correct why did you delete?

Comment (at-trungnguyen) [reply app/src/at-thanhhoang/java/asiantech/internship/summer/rest_api/models/ImageInfo.java]: `@SerializedName` use when `key` wrong convention.

**Comment by at-quocnguyen** [comment]
Resolved Conflict @at-thanhhoang
